utils.ts

```diff
@@ -138,16 +139,20 @@ export const getUSDCToUSDGLOUniswapDeeplink = (
   }
 
   const inputCurrency = getUSDCContractAddress(chain);
-  let outputCurrency, uniswapChain;
+  let outputCurrency, swapChain;
   if (chain.id === mainnet.id || chain.id === goerli.id) {
     outputCurrency = getSmartContractAddress(mainnet.id);
-    uniswapChain = "mainnet";
+    swapChain = dex === "Uniswap" ? "mainnet" : "ethereum";
   } else {
     outputCurrency = getSmartContractAddress(polygon.id);
-    uniswapChain = "polygon";
+    swapChain = "polygon";
   }
 
-  return `https://app.uniswap.org/#/swap?inputCurrency=${inputCurrency}&outputCurrency=${outputCurrency}&exactAmount=${amount}&exactField=input&chain=${uniswapChain}`;
+  const outputUrl =
+    dex === "Matcha"
+      ? `https://matcha.xyz/tokens/${swapChain}/${outputCurrency}`
```

Comment:
Not sure this is deep linked enough?

Reply:
I tried adding both input and output currencies like 

`https://matcha.xyz/tokens/polygon/0x2791bca1f2de4661ed88a30c99a7a9449aa84174/0x4f604735c1cf31399c6e711d5962b2b3e0225ad3` but it doesn't work :(